### PR TITLE
Upgrade to version 6

### PR DESCRIPTION
**Pull request by ramyferjaniadot**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/adotBidAdapter.js [hunk 1]
```diff
@@ -1,313 +1,236 @@
...
-// eslint-disable-next-line no-template-curly-in-string
-const AUCTION_PRICE = '${AUCTION_PRICE}';
-const TTL = 10;
-
-const SUPPORTED_VIDEO_CONTEXTS = ['instream', 'outstream'];
-const SUPPORTED_INSTREAM_CONTEXTS = ['pre-roll', 'mid-roll', 'post-roll'];
-const SUPPORTED_VIDEO_MIMES = ['video/mp4'];
-const BID_SUPPORTED_MEDIA_TYPES = ['banner', 'video', 'native'];
-
+const REQUIERED_VIDEO_PARAMS = ['mimes', 'minduration', 'maxduration', 'protocols'];
```
Comment (ako-adot) [inline modules/adotBidAdapter.js]:
```suggestion
const REQUIRED_VIDEO_PARAMS = ['mimes', 'minduration', 'maxduration', 'protocols'];
```

File: modules/adotBidAdapter.js [hunk 2]
```diff
@@ -319,249 +242,163 @@ function generateNativeFromAdUnit(data) {
...
+  return !video || isValidVideo(video);
+}
+
+/**
+ * Build OpenRtb request from Prebid AdUnits and Bidder request
+ *
+ * @param {Array<AdUnit>} adUnits Array of PrebidJS Adunit
+ * @param {BidderRequest} bidderRequest PrebidJS BidderRequest
+ * @param {string} requestId Request ID
+ * @returns
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]: jsdoc

File: modules/adotBidAdapter.js [hunk 3]
```diff
@@ -319,249 +242,163 @@ function generateNativeFromAdUnit(data) {
...
+  const bidderUrl = BIDDER_URL.replace('{PUBLISHER_PATH}', publisherPath);
+
+  return Object.keys(adUnits).map((requestId) => buildAjaxRequest(adUnits[requestId], bidderRequest, bidderUrl, requestId));
+}
+
+/**
+ * Build Native PrebidJS Response grom OpenRtb Response
+ *
+ * @param {OpenRtbBid} bid
+ * @returns {}
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]: jsdoc

File: modules/adotBidAdapter.js [hunk 4]
```diff
@@ -641,54 +459,179 @@ function buildOutstreamRenderer(impressionData) {
...
-        return length > 0 && group.values[length - 1]
-      });
+/**
+ * Return if given bid and imp are valid
+ *
+ * @param {OpenRtbBid} bid OpenRtb Bid
+ * @param {Imp} imp OpenRtb Imp
+ * @returns {boolean}
+ */
+function bidImpIsInvalid(bid, imp) {
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]:
```suggestion
function isBidImpInvalid(bid, imp) {
```

File: modules/adotBidAdapter.js [hunk 5]
```diff
@@ -641,54 +459,179 @@ function buildOutstreamRenderer(impressionData) {
...
-      .reduce((acc, seatBid) => acc.concat(seatBidsToAds(seatBid, bidResponse, serverRequest)), []);
-  }
+/**
+ * Call Adunit getFloor function with given argument to get specific floor.
+ * Return 0 by default
+ *
+ * @param {AdUnit} adUnit
+ * @param {Array<number>|stirng} size Adunit size or *
+ * @param {string} mediaType
+ * @param {stirng} currency USD by default
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]:
```suggestion
 * @param {string} currency USD by default
```

File: modules/adotBidAdapter.js [hunk 6]
```diff
@@ -641,54 +459,179 @@ function buildOutstreamRenderer(impressionData) {
...
-    return bidResponse.seatbid
-      .filter(seatBid => isPlainObject(seatBid) && isArray(seatBid.bid))
-      .reduce((acc, seatBid) => acc.concat(seatBidsToAds(seatBid, bidResponse, serverRequest)), []);
-  }
+/**
+ * Call Adunit getFloor function with given argument to get specific floor.
+ * Return 0 by default
+ *
+ * @param {AdUnit} adUnit
+ * @param {Array<number>|stirng} size Adunit size or *
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]:
```suggestion
 * @param {Array<number>|string} size Adunit size or *
```

File: modules/adotBidAdapter.js [hunk 7]
```diff
@@ -641,54 +459,179 @@ function buildOutstreamRenderer(impressionData) {
...
+ * Call getFloor for each format and return the lower floor
+ * Return 0 by default
+ *
+ * interface Format { w: number; h: number }
+ *
+ * @param {AdUnit} adUnit
+ * @param {Array<Format>} formats Media formats
+ * @param {string} mediaType
+ * @param {stirng} currency USD by default
+ * @returns
```
Comment (ramyferjaniadot) [inline modules/adotBidAdapter.js]: jsdoc

File: modules/adotBidAdapter.js [hunk 8]
```diff
@@ -319,249 +242,163 @@ function generateNativeFromAdUnit(data) {
...
-      consent: adUnitContext.gdprConsent.consentString
-    }
-  };
-}
-
-function getRegulationFromAdUnitContext(adUnitContext) {
-  if (!adUnitContext || !adUnitContext.gdprConsent) return null;
-  if (!isBoolean(adUnitContext.gdprConsent.gdprApplies)) return null;
+  const media = IMP_BUILDER[mediaType](mediaTypes[mediaType], bidderRequest, adUnit)
+  const currency = config.getConfig('currency.adServerCurrency') || 'USD';
```
Comment (ako-adot) [inline modules/adotBidAdapter.js]: Tu pourrais mettre une constante DEFAULT_CURRENCY

File: modules/adotBidAdapter.js [hunk 9]
```diff
@@ -641,54 +459,179 @@ function buildOutstreamRenderer(impressionData) {
...
+/**
+ * Call getFloor for each format and return the lower floor
+ * Return 0 by default
+ *
+ * interface Format { w: number; h: number }
+ *
+ * @param {AdUnit} adUnit
+ * @param {Array<Format>} formats Media formats
+ * @param {string} mediaType
+ * @param {stirng} currency USD by default
```
Comment (ako-adot) [inline modules/adotBidAdapter.js]:
```suggestion
 * @param {string} currency USD by default
```

File: modules/adotBidAdapter.js (hunk 5 above)
Comment (ako-adot) [inline modules/adotBidAdapter.js]:
```suggestion
 * @param {string} currency USD by default
```

File: modules/adotBidAdapter.js (hunk 6 above)
Comment (ako-adot) [inline modules/adotBidAdapter.js]:
```suggestion
 * @param {Array<number>|string} size Adunit size or *
```

**Review by ako-adot — APPROVED** [review]

**Review by verpilc — APPROVED** [review]

**Pull request merged**